internal/rootcoord/meta_table.go

```diff
@@ -448,6 +450,30 @@ func (mt *MetaTable) AlterCollection(ctx context.Context, oldColl *model.Collect
 	return nil
 }
 
+func (mt *MetaTable) RenameCollection(ctx context.Context, oldName string, newName string, ts Timestamp) error {
+	mt.ddLock.RLock()
+	defer mt.ddLock.RUnlock()
+	ctx = contextutil.WithTenantID(ctx, Params.CommonCfg.ClusterName.GetValue())
+
+	oldColl, err := mt.GetCollectionByName(ctx, oldName, ts, true)
+	if err != nil {
+		return err
+	}
+
+	newColl := oldColl.Clone()
+	newColl.Name = newName
```

Comment:
We didn't check if `newName` is valid here:
- Whether `newName` satisfies the rule of collection?
- Whether `newName` exists or it's an alias of collection?